### DISCUSSION TITLE
Full nitro support

### DISCUSSION THREAD

**Discussion by anuragkumar19** [nksaraf/vinxi#347] [Ideas]
Vinxi is based nitro, Nitro has many features for api development like tasks, websocket, kv storage, database, cache etc.
 
I think we should add method to dedicate a directory (`/server`) to nitro where we can access all nitro features.

**Comment by ryansolid** [comment]
The problem is if Vinxi exposes a convention it goes against the philosophy of the project. In a sense the worst part of Vite is `index.html`. You can't choose where it goes and there is very little to be done with it. It's one thing but it takes away from Vite's ability to configure anything the way you want. It's easy enough to sidestep it though.

The worst part of Nitro is the the `/server` directory. It makes convention decisions that should be up to the framework. I tried to get my head into it especially as it would save us work. But it's the wrong thing to do.

Like if Nitro didn't push towards the `/server` directory and when Vite 6 comes out, maybe we don't even need Vinxi to exist. But that's the baseline I think for an acceptable solution here.
